pkg/controller/hypershift/hostedcluster_controller.go

```diff
@@ -0,0 +1,387 @@
+	//  ../../../cmd/manager/main.go:100
+	//
+	//	if err := c.Watch(
+	//		source.NewImportSecretSource(importSecretInformer),
+	//		&source.ManagedClusterSecretEventHandler{},
+	//		predicate.Predicate(predicate.Funcs{
+	//			GenericFunc: func(e event.GenericEvent) bool { return false },
+	//			DeleteFunc:  func(e event.DeleteEvent) bool { return false },
+	//			CreateFunc:  func(e event.CreateEvent) bool { return true },
+	//			UpdateFunc: func(e event.UpdateEvent) bool {
+	//				new, okNew := e.ObjectNew.(*corev1.Secret)
+	//				old, okOld := e.ObjectOld.(*corev1.Secret)
+	//				if okNew && okOld {
+	//					return !equality.Semantic.DeepEqual(old.Data, new.Data)
+	//				}
+	//
+	//				return false
+	//			},
+	//		}),
+	//	); err != nil {
+	//		return err
+	//	}
+
+	return nil
+}
+
+// ReconcileHostedcluster reconciles the hostedcluster that is in the hosted cluster namespace
+// to import the managed cluster
+type ReconcileHostedcluster struct {
+	client     client.Client
+	kubeClient kubernetes.Interface
+	recorder   events.Recorder
+}
+
+// blank assignment to verify that ReconcileHostedcluster implements reconcile.Reconciler
+var _ reconcile.Reconciler = &ReconcileHostedcluster{}
+
+// Reconcile the hostedcluster that is in the managed cluster namespace to import the managed cluster.
+//
+// Note: The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileHostedcluster) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request", request)
+	reqLogger.Info("Reconciling HostedCluster")
+	defer reqLogger.Info("Reocileing HostedCluster Done")
+
+	clusterName := request.Name
+	hCluster := &hyperv1.HostedCluster{}
+	err := r.client.Get(ctx, request.NamespacedName, hCluster)
+	if errors.IsNotFound(err) {
+		return reconcile.Result{}, nil
+	}
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	if !hCluster.DeletionTimestamp.IsZero() {
+		// the hostedcluster is deleting, its managed cluster may already be detached (the managed cluster has been deleted,
+		// but the namespace is remained), if it has import finalizer, we remove its namespace
+		return reconcile.Result{}, r.removeImportFinalizer(ctx, hCluster)
+	}
+
+	// TODO: check if UI would do so or not
+	if err := r.ensureManagedClusterCR(ctx, request.NamespacedName); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// add a managed cluster finalizer to the hosted cluster, to handle the managed cluster detach case.
+	if err := r.addClusterImportFinalizer(ctx, hCluster); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	// if there is an auto import secret in the managed cluster namespce, we will use the auto import secret to import the cluster
+	_, err = r.kubeClient.CoreV1().Secrets(clusterName).Get(ctx, constants.AutoImportSecretName, metav1.GetOptions{})
```

Comment:
I don't think we need auto-import secret in this scenario.  kubeconfig of hostedCluster.status is enough.

Reply:
I read the following:
https://access.redhat.com/documentation/en-us/red_hat_advanced_cluster_management_for_kubernetes/2.4/html/clusters/managing-your-clusters#importing-a-managed-cluster-with-the-cli

at section: `1.8.2.4. Importing the cluster`, it mentioned the auto-import secret stuff... do we need to worry about this for hypershift?

If not, we can remove the above section.

If so, we might want to have a logic to priorities, the auto-import secret and the auto-generated secret by `hostedCluster.Status.Kubeconfig`

Reply:
if user create a auto-import secret for the managedCluster, the autoimport controller will handle that, it is duplicate here.